django_cte/cte.py

```diff
@@ -125,7 +125,20 @@ class CTEManager(Manager):
     """Manager for models that perform CTE queries"""
 
     def get_queryset(self):
-        return CTEQuerySet(self.model, using=self._db)
+        qs = super(CTEManager, self).get_queryset()
+        if isinstance(qs, CTEQuerySet):
+            return qs
+        else:
+            assert type(qs) == QuerySet, \
+                "models with CTE support need to use a CTEQuerySet"
```

Comment:
What do you think of changing this to raise `TypeError` instead of `AssertionError`? It seems like a person using django-cte could hit this error if they configured or used the library incorrectly. Assertion errors should only ever be triggered by bugs in the library code itself.